admin/src/components/generator/ContentGeneratorPanel.jsx

```diff
@@ -0,0 +1,482 @@
+	const promptTemplateSelections = Object.entries( promptTemplates ).reduce( ( acc, [ key, value ] ) => {
+		acc[ key ] = value.name;
+		return acc;
+	}, {} );
+	const [ postTypes, setPostTypes ] = useState( {} );
+	const [ postType, setPostType ] = useState( 'post' );
+	const [ title, setTitle ] = useState( '' );
+	const [ urlsList, setUrlsList ] = useState( [] );
+	const [ keywordsList, setKeywordsList ] = useState( [] );
+	const [ serpUrlList, setSerpUrlList ] = useState( [] );
+	const [ domain, setDomain ] = useState( '' );
+	const [ semanticContext, setSemanticContext ] = useState( '' );
+	const [ selectedPromptTemplate, setSelectedPromptTemplate ] = useState( '0' );
+	const [ promptVal, setPromptVal ] = useState( '' );
+	const [ modelName, setModelName ] = useState( 'gpt-3.5-turbo' );
+	const [ isGenerating, setIsGenerating ] = useState( false );
+	const [ errorGeneration, setErrorGeneration ] = useState( '' );
+	const [ generatedPostLink, setGeneratedPostLink ] = useState( '' );
+
+	const contextTypes = {
+		NO_CONTEXT: 'No Data Source',
+		URL_CONTEXT: 'URL Data Source',
+		DOMAIN_CONTEXT: 'Domain Data Source',
+		SERP_CONTEXT: 'Google Search Source',
+	};
+
+	const contextTypesDescription = {
+		NO_CONTEXT: __( 'If no data source is selected, the content will be generated solely based on the prompts you provide. No additional context or supplemental data will be implemented in the creation process.' ),
+		URL_CONTEXT: __( 'When using a URL as the data Source, the generated content will be influenced by the information found in the chosen URL(s). This allows the text to be as relevant as possible to the content on your selected page(s).' ),
+		DOMAIN_CONTEXT: __( 'Using a domain as the data source means the text will be based on data collected from pages across your entire domain. To use this feature, you\'ll need to add your domain to the domain section.' ),
+		SERP_CONTEXT: __( 'Opting for a Google Search means that the content generated will focus on the Primary Keyword, similar keywords, and Search Engine Results Page (SERP) data. We collect information from top websites linked to your keyword to create original and relevant content.' ),
+	};
+
+	const contextTypePromptPlaceholder = {
+		NO_CONTEXT: __( 'Your prompt to be used for generating text…' ),
+		URL_CONTEXT: __( 'The prompt to be used for generating text from each url…' ),
+		DOMAIN_CONTEXT: __( 'The prompt to be used to generate text from relevant content in your whole domain…' ),
+		SERP_CONTEXT: __( 'The prompt to be used to generate text from top SERP Results targeting your keyword…' ),
+	};
+
+	useEffect( () => {
+		getPostTypes().then( async ( res ) => {
+			const rsp = await res.json();
+			setPostTypes( rsp );
+		} );
+	}, [] );
+
+	const handleCreatePost = async () => {
+		createPost( editorVal, title, postType ).then( async ( res ) => {
+			const rsp = await res.json();
+			setGeneratedPostLink( decodeURIComponent( rsp.edit_post_link ) );
```

Comment:
why not save results with tanstack query as separate query?

Reply:
This request is for creating a new post, it doesn't make sense to cache this